connector-j-5/src/main/java/com/google/cloud/sql/mysql/SocketFactory.java

```diff
@@ -31,8 +31,24 @@
 
   private Socket socket;
 
+  private static final String APPLICATION_NAME = "mysql-socket-factory-connector-j-5";
+
+  private static String getUserAgentString() {
+    try {
+      Properties packageInfo = new Properties();
+      packageInfo
+          .load(SocketFactory.class.getClassLoader().getResourceAsStream("project.properties"));
```

Comment:
The gcp-spring folks are getting the artifact version without having to read from a file ([example](https://github.com/spring-cloud/spring-cloud-gcp/blob/8850186d11d9604ac1ce771a5313389306fa3350/spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java#L173-L174)). Can we do something similar here? 

Looks like there is a [`getImplementationtitle`](https://docs.oracle.com/javase/8/docs/api/java/lang/Package.html#getImplementationTitle--) that might be helpful as well.

Reply:
I tried that earlier but kept getting null values for the version. From looking it up, I gathered it was because Maven jar builds don't include that information in the jar manifest by default. I added a config option to the parent pom.xml which added that metadata to the jar, but it still showed up as null when getting the value.